src/game/system/_private/PhysicsSystem.cpp

```diff
@@ -1,24 +1,216 @@
 #include "PhysicsSystem.h"
+#include "TransformComponent.h"
+#include "MovableComponent.h"
+
+
+bool PhysicsSystem::s_IsWorldInitialized = false;
 
 PhysicsSystem::PhysicsSystem() :
 	m_pPhysicsComponents(EntityManager::GetAll<PhysicsComponent>())
 {
+	
 }
 
+
+
 bool PhysicsSystem::Initialize()
 {
+	if (!s_IsWorldInitialized)
+	{
+		m_pBroadphase = new btDbvtBroadphase();
```

Comment:
This works fine, but I meant that the static bool would be used to initialize the static Bullet objects (since we wouldn't want the PhysicsSystem constructor to `new` objects that have already been created). It definitely seems like we can't beat the efficiency of having these things be static (but it is entirely up to you).